runtime/js-compute-runtime/fastly.wit

```diff
@@ -553,17 +334,293 @@ interface fastly {
     /// this client is requested to insert or revalidate an object
     must-insert-or-update,
   }
+}
+
+/*
+ * Fastly Log
+ */
+interface log {
+
+  use types.{error, log-endpoint-handle}
+
+  endpoint-get: func(name: string) -> result<log-endpoint-handle, error>
+
+  write: func(h: log-endpoint-handle, msg: string) -> result<_, error>
+}
+
+/*
+ * Fastly Dictionary
+ */
+interface dictionary {
+
+  use types.{error, dictionary-handle}
+
+  open: func(name: string) -> result<dictionary-handle, error>
+
+  get: func(h: dictionary-handle, key: string) -> result<option<string>, error>
+}
+
+/*
+ * Fastly Secret Store
+ */
+interface secret-store {
+
+  use types.{error, secret-store-handle, secret-handle}
+
+  open: func(name: string) -> result<secret-store-handle, error>
+
+  get: func(store: secret-store-handle, key: string) -> result<option<secret-handle>, error>
+
+  plaintext: func(secret: secret-handle) -> result<option<string>, error>
+
+}
+
+/*
+ * Fastly KV Store
+ */
+interface object-store {
+
+  use types.{object-store-handle, fd, body-handle, error}
+
+  open: func(name: string) -> result<object-store-handle, error>
+
+  lookup: func(store: object-store-handle, key: string) -> result<option<body-handle>, error>
+
+  lookup-as-fd: func(store: object-store-handle, key: string) -> result<option<fd>, error>
+
+  // Should object store insert return "inserted" bool?
+  insert: func(store: object-store-handle, key: string, body-handle: body-handle) -> result<_, error>
+
+}
+
+/*
+ * Fastly HTTP Body
+ */
+interface http-body {
+  use types.{error, body-handle, body-write-end}
+
+  new: func() -> result<body-handle, error>
+
+  append: func(dest: body-handle, src: body-handle) -> result<_, error>
+
+  read: func(h: body-handle, chunk-size: u32) -> result<list<u8>, error>
+
+  write: func(h: body-handle, buf: list<u8>, end: body-write-end) -> result<u32, error>
+
+  close: func(h: body-handle) -> result<_, error>
+}
+
+/*
+ * Fastly HTTP Req
+ */
+interface http-req {
+
+  use types.{
+    request-handle, http-cache-override-tag, http-version, body-handle, error,
+    response, pending-request-handle, content-encodings, framing-headers-mode,
+    dynamic-backend-config
+  }
```

Comment:
When it comes to type sharing, the WASI approach is to have a subsystem specific interface - eg:

```
interface http {
  type request-handle = u32 // etc
}

interface http-req {
  use http.{request-handle}
}
```

Or even call it `http-types` directly.

Reply:
That works for me!